SwiftWisdom/Core/UIKit/UILabel+Typography.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  UILabel+Typography.swift
+//  SwiftWisdom
+//
+//  Created by Patrick Butkiewicz on 4/15/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+extension UILabel {
+    
+    /**
+     Sets the text of the UILabel instance to an attributed string with tracking applied
+     - Parameter value: The value in points that each character is kerned to
+     */
+    
+    public func ip_setCharacterSpacing(value: CGFloat) {
+        addAttribute(NSKernAttributeName, value: value)
+    }
+    
+    /**
+     Sets the text of the UILabel instance to an attributed string with the specified line spacing
+     - Parameter value: The space, in points, between each line.
+     */
+    
+    public func ip_setLineSpacing(value: CGFloat) {
+        let paragraphSpacing = NSMutableParagraphStyle()
+        paragraphSpacing.lineSpacing = value
+        addAttribute(NSParagraphStyleAttributeName, value: paragraphSpacing)
+    }
+    
+    // MARK: Private
+    
+    private func addAttribute(attr: String, value: AnyObject) {
+        let attributedText = mutableAttributedText()
+        attributedText.addAttributes([attr:value], range: NSMakeRange(0, attributedText.length))
+        self.text = nil
+        self.attributedText = attributedText
+    }
+    
+    private func baseAttributes() -> [String:AnyObject] {
+        return [NSFontAttributeName : self.font, NSForegroundColorAttributeName : self.textColor]
```

Comment:
This leaves out alignment, right? Maybe make an NSParagraphStyle to capture that?
```
let newParagraphStyle = NSMutableParagraphStyle()
newParagraphStyle.alignment = textAlignment
return [
   NSFontAttributeName : self.font, 
   NSForegroundColorAttributeName : self.textColor, 
   NSParagraphStyleAttributeName : newParagraphStyle
]
```